library/std/src/process/tests.rs

```diff
@@ -417,6 +417,44 @@ fn env_empty() {
     assert!(p.is_ok());
 }
 
+#[test]
+#[cfg(not(windows))]
+#[cfg_attr(any(target_os = "emscripten", target_env = "sgx"), ignore)]
+fn main() {
+    const PIDFD: &'static str =
+        if cfg!(target_os = "linux") { ", create_pidfd: false" } else { "" };
+
+    let mut command = Command::new("some-boring-name");
+
+    assert_eq!(
+        format!("{command:?}"),
+        format!(
+            r#"Command {{ program: "some-boring-name", args: ["some-boring-name"], env: CommandEnv {{ clear: false, saw_path: false, vars: {{}} }}, cwd: None, uid: None, gid: None, saw_nul: false, groups: None, stdin: None, stdout: None, stderr: None, pgroup: None{PIDFD} }}
+Command line: "some-boring-name""#
+        )
+    );
+
+    command.args(&["1", "2", "3"]);
+
+    assert_eq!(
+        format!("{command:?}"),
+        format!(
+            r#"Command {{ program: "some-boring-name", args: ["some-boring-name", "1", "2", "3"], env: CommandEnv {{ clear: false, saw_path: false, vars: {{}} }}, cwd: None, uid: None, gid: None, saw_nul: false, groups: None, stdin: None, stdout: None, stderr: None, pgroup: None{PIDFD} }}
+Command line: "some-boring-name" "1" "2" "3""#
```

Comment:
The quotes here suggest that this is shell-safe. But we're not doing any shell-escaping. Passing arguments as strings can be perfectly safe, depending on the target application, but when pasted to a shell they can be dangerous.

E.g. `Command::new("mysandbox").arg("$(echo 'pwned' > /dev/tty)")` would debug-print as
`Command line: "mysandbox" "$(echo 'pwned' > /dev/tty)"` 

When you paste that into various shells you get pwned before the sandbox is executed.

Now it's probably very unlikely that people just copy-paste debug output containing untrusted parameters but that quoting does give a bit of a false impression.

And it's not just malicious inputs, it's also random shell metacharacters breaking the commandline. Using single quotes would improve this somewhat but it still wouldn't be watertight because blackslashes and single quotes within the string itself could break out of that.

Reply:
I would be in favour of dropping command-line to avoid these issues, see comment above.

Reply:
This is pre-existing behavior so I won't block the PR on you fixing it. I don't think it should be removed.

Reply:
Have not addressed escaping because had no idea how to tackle it